pkg/controller/handler.go

```diff
@@ -227,6 +229,31 @@ func (p *Propeller) Handle(ctx context.Context, namespace, name string) error {
 				ResetFinalizers(mutatedWf)
 			}
 		}
+
+		// ExecutionNotFound error is returned when flyteadmin is missing the workflow. This is not
+		// a valid state unless we are experiencing a race condition where the workflow has not yet
+		// been inserted into the db (ie. workflow phase is WorkflowPhaseReady).
+		if err != nil {
+			e, ok := err.(*errors.WorkflowErrorWithCause)
```

Comment:
I would run the error through `stdErrors.Unwrap` first... in case it was wrapped somewhere on the stack...
```
import (
...
   stdErrors "errors"
)
```

Reply:
This is a very thrifty API and a very useful suggestion. It looks like errors.Unwrap function only unwraps a single layer though (and requires the Unwrap function defined on the internal error). Alternatively, I changed the check to use errors.Is directly on the ExecutionNotFound error which checks if it is anywhere in the stack. However, this does require a slight modification to the flyteidl repo to include EventError.Is and EventError.Unwrap functions. I will submit a pull request shortly.